libvast/vast/system/partition.hpp

```diff
@@ -149,7 +149,7 @@ struct passive_partition_state {
   vast::chunk_ptr partition_chunk;
 
   /// A typed view into the `partition_chunk`.
-  const fbs::Partition* flatbuffer;
+  const fbs::v1::Partition* flatbuffer;
```

Comment:
I think I would try to keep the version part in the implementation, at least for the reading side. It would be really desirable to be able to load a `vast.db` from the previous version of vast if the change between `N - 1` and `N` permits it.

Reply:
After some offline discussion, we agreed that this makes more sense the other way round:

For `pack()` we can return the union type since there's a unique embedding from `partition::v0` into `Partition`, and the caller always performs this work anyways.

For `unpack()` it depends on the exact changes between the versions whether we can trivially unpack a different version, want to emit a warning, require some explicit conversion or just refuse it altogether, so the calling code should just do whatever it wants to do beforehand so the unpack function can focus purely on unpacking the data.